crates/engine/src/lib.rs

```diff
@@ -61,43 +61,55 @@ pub struct RuntimeContext<T> {
     pub data: Option<T>,
 }
 
+/// The engine struct that encapsulate wasmtime engine
+#[derive(Clone)]
+pub struct Engine(wasmtime::Engine);
+
+impl Engine {
+    /// Create a new engine and initialize it with the given config.
+    pub fn new(mut config: wasmtime::Config) -> Result<Self> {
+        // In order for Wasmtime to run WebAssembly components, multi memory
+        // and module linking must always be enabled.
+        // See https://github.com/bytecodealliance/wit-bindgen/blob/main/crates/wasmlink.
+        config.wasm_multi_memory(true);
+        config.wasm_module_linking(true);
+        Ok(Self(wasmtime::Engine::new(&config)?))
+    }
+
+    /// Get a clone of the internal `wasmtime::Engine`.
+    /// WARNING: The configuration of this Engine is likely to change in the future, and
+    /// will not be covered by any future stability guarantees.
+    pub fn inner(&self) -> wasmtime::Engine {
```

Comment:
It doesn't matter much for a `wasmtime::Engine`, but in _general_ you would return a reference here and let the caller `.clone`

Reply:
good tip!